src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/appendoverwrite/factory/entity/PendingObservation.java

```diff
@@ -115,13 +113,15 @@ public <U> List<U> brapiPut(List<U> members) throws ApiException, IllegalArgumen
             return new ArrayList<U>();
         }
 
+        Map<String, BrAPIObservation> mutatedObservationByDbId = new HashMap<>();
         List<BrAPIObservation> updatedObservations = new ArrayList<>();
         for (U member : members) {
             BrAPIObservation observation = (BrAPIObservation) member;
-            Optional.ofNullable(brAPIObservationDAO.updateBrAPIObservation(observation.getObservationDbId(), observation, importContext.getProgram().getId())).ifPresent(updatedObservations::add);
+            mutatedObservationByDbId.put(observation.getObservationDbId(), observation);
+//            Optional.ofNullable(brAPIObservationDAO.updateBrAPIObservation(observation.getObservationDbId(), observation, importContext.getProgram().getId())).ifPresent(updatedObservations::add);
```

Comment:
Looks good. I would just delete all of this.

Reply:
I may be misunderstanding your comment, but _mutatedObservationByDbId_ is used on line 124.  If I delete lines 116-121 it will not be declared or populated.

Reply:
Thanks!  I received your full comment in Slack and have implemented your suggestions.